site/docs/id/guide/filter-queries.md

```diff
@@ -350,8 +350,23 @@ Selain itu, kamu juga akan diberikan pesan error yang lebih bermanfaat.
 Validasi filter query hanya dilakukan sekali, ketika bot diinisialisasi dan `bot.on()` dipanggil.
 
 Ketika dimulai, grammY menurunkan _function predicate_ dari filter query dengan cara memecahnya menjadi beberapa komponen query.
-Setiap komponen akan di-map ke sebuah function yang mengerjakan satu pemeriksaan `in`, atau dua pemeriksaan jika komponen tersebut diabaikan dan dua value perlu dilakukan pemeriksaan.
+Setiap komponen akan di-map ke sebuah function yang mengerjakan satu pemeriksaan truthiness untuk properti objek, atau dua pemeriksaan jika komponen tersebut diabaikan dan dua value perlu dilakukan pemeriksaan.
 Function-function ini kemudian disatukan untuk membentuk sebuah predicate yang akan memeriksa sebanyak mungkin value yang relevan untuk query, tanpa melakukan proses perulangan terhadap key object `Update`.
 
 Sistem ini menggunakan lebih sedikit operasi dibandingkan dengan beberapa library lainnya, dimana dibutuhkan beberapa pengecekan array ketika melakukan routing update.
 Ini membuat sistem filter query grammY selain lebih unggul juga jauh lebih cepat.
+
+### Type Safety
+
+Sebagaimana yang disebutkan diatas, filter queries secara otomatis akan mempersempit properti tertentu pada objek context dan menghasilkan satu atau lebih predicate dalam fungsi predicate type di TypeScript.
+Secara umum, kamu dapat mempercayai type inference itu ia selalu bekerja dengan benar.
+Jika suatu properti type infer-nya tersedia, kamu dapat dapat menggunakan type tersebut dengan aman.
+Jika suatu properti type infer-nya hilang, berarti ada kasus tertentu yang membuat type-nya hilang.
+Bukan ide yang baik untuk melakukan casting type dengan operator `!`.
+
+> Mungkin ini kurang jelas tentang apa maksud kasus tersebut.
+> Jangan ragu untuk bertanya di [obrolan grup](https://t.me/grammyjs) jika kamu tidak memahaminya.
+
+Membuat type ini merumitkan.
+Banyak sekali pengetahuan tentang API Bot yang masuk pada bagian grammY ini.
+Apabila kamu ingin lebih memahami tentang pendekatan dasar bagaimana type ini dibuat, berikut ada [pembahasannya di Youtube](https://youtu.be/ZvT_xexjnMk) yang mungkin bisa kamu tonton.
```

Comment:
```suggestion
Apabila kamu ingin lebih memahami tentang pendekatan dasar bagaimana type ini dibuat, berikut ada [pembahasannya di Youtube](https://youtu.be/ZvT_xexjnMk) yang mungkin bisa kamu tonton.

::: tip
Catatan penerjemah :

Pembahasan pada video di atas dalam berbahasa Inggris.
Apabila kamu merasa kurang jelas/kesulitan dalam memahami penjelasan tersebut, gunakanlah fitur subtitle dan terjemahan otomatisnya.
:::
```

Reply:
```suggestion
Apabila kamu ingin memahami lebih dalam bagaimana type ini dibuat, berikut pembahasan lengkapnya yang bisa kamu [tonton di Youtube](https://youtu.be/ZvT_xexjnMk).
```

Reply:
I don't see the Indonesian subtitle available for that video (yet), so it's not relevant to suggest the reader to turn on the subtitle.

Reply:
It should be
```suggestion
Apabila kamu ingin lebih memahami tentang pendekatan dasar bagaimana type ini dibuat, berikut ada [pembahasannya di YouTube](https://youtu.be/ZvT_xexjnMk) yang mungkin bisa kamu tonton.
```
right?

Reply:
I'm working on the subtitles

Reply:
> I'm working on the subtitles

Okay. We will leave the tips then. Do you need help to generate ID subtitle as well?

Reply:
I think that @niusia-ua is right in https://github.com/grammyjs/website/pull/1074#discussion_r1642602223 so it isn't needed, but if the translation is bad, you can contribute it later